.github/workflows/tf-cleanup.yaml

```diff
@@ -74,7 +77,7 @@ jobs:
               eval "${{ inputs.tf_pre_run }}"
             fi
         with:
-          workspace: ${{ github.head_ref }}
+          workspace: ${{ inputs.tf_workspace || github.head_ref }}
```

Comment:
How do we know the value to give to tf_workspace ? Is it not possible, when manually triggering a workflow to specify the branch and then having it in github.ref ?

Reply:
At a time I had this table: 
```
Set ENVIRONMENT based on:

 | github.event_name | github.ref_name | github.base_ref | ENVIRONMENT |
 | ----------------- | --------------- | --------------- | ----------- |
 | workflow_dispatch | main            | (not set)       | prod        |
 | workflow_dispatch | FEAT_BRANCH     | (not set)       | error       | *)
 | pull_request      | #/merge         | main            | dev         |
 | pull_request      | #/merge         | FEAT_BRANCH     | error       | *)
 | push              | main            | (not set)       | dev         |
 | push              | FEAT_BRANCH     | (not set)       | error       | *)
 *) Worlflow not called because of filter on branch

Note:
  - If triggered by "push" (when PR is merged), branch is ${{github.ref_name}} and ${{ github.base_ref }} is not defined.
  - If triggered by "pull_request" (when commit on feature branch), branch is ${{ github.base_ref }} and ${{ github.ref_name }} contains PR number/merge.
  - If triggered by "workflow_dispatch" (manual trigger of the WF), branch is ${{ github.ref_name }} and ${{ github.base_ref }} is not defined.



 environment: ${{ (github.event_name == 'workflow_dispatch' && github.ref_name == 'main' && 'prod') || (github.base_ref || github.ref_name ) == 'main' && 'dev' || 'error' }}
```

Reply:
> Is it not possible, when manually triggering a workflow to specify the branch and then having it in github.ref ?

Thanks for the comment, yes it is possible, but it will be done when calling the workflow itself, here is an example: https://github.com/tx-pts-dai/terraform-aws-kubernetes-platform/blob/dont-run-examples-deployments-on-renovates-pr/.github/workflows/cleanup.yaml#L24

I would not hard code here the reference, since this works for workflow_dispatch started by selecting a branch name, but for other triggers you might want to pass the workspace name differently.